hy/compiler.py

```diff
@@ -1216,20 +1228,72 @@ def _compile_import(expr, module, names=None, importer=asty.Import):
 
         return rimports
 
-    @builds("get")
-    @checkargs(min=2)
-    def compile_index_expression(self, expr):
-        expr.pop(0)  # index
+    def _compile_subscript(self, root, value, exprs):
+        dims = []
+        ret = Result()
+
+        for expr in exprs:
+            if is_builtin("slice", expr):
+                if len(expr) > 4:
+                    raise HyTypeError(expr, "slice expected at most "
+                                            "3 arguments")
+
+                nodes = [None] * 3
+                for i, e in enumerate(expr[1:]):
+                    ret += self.compile(e)
+                    nodes[i] = ret.force_expr
+
+                dims.append(
+                    ast.Slice(lower=nodes[0], upper=nodes[1], step=nodes[2]))
+            elif not PY3 and isinstance(expr, HySymbol) and expr == "...":
+                dims.append(ast.Ellipsis())
+            else:
+                ret += self.compile(expr)
+                dims.append(ret.force_expr)
+
+        def box_as_slice(expr):
+            if isinstance(expr, ast.slice):
+                return expr
+            return ast.Index(value=expr)
+
+        # How the subscript is constructed will depend on if any of
+        # the components subclass ast,slice. If they do, then we have
+        # to construct an ExtSlice instead of a Tuple.
```

Comment:
Is there any reason not to use ExtSlice instead of a tuple? That would simplify things a bit.

Reply:
Not sure, but I suspect its fine. I figured it made the most send to match exactly what Python does.

Reply:
Oh, to expand on this, its probably worth at least merging the identical behaviour. If a simplification works, that's awesome, but that can be added next and at least there's always something more technically correct to revert to.

Reply:
Okay.